backend/s3/s3.go

```diff
@@ -0,0 +1,200 @@
+package s3
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"strings"
+	"sync"
+
+	"github.com/mitchellh/goamz/aws"
+	as3 "github.com/mitchellh/goamz/s3"
```

Comment:
This was just a little internal weirdness to me to use a package s3 while writing package s3. I'll take it off.